src/ILLink.Shared/TrimAnalysis/SystemTypeValue.cs

```diff
@@ -19,11 +21,33 @@ record SystemTypeValue : SingleValue
 		public override string ToString () => this.ValueToString (RepresentedType);
 	}
 
-	sealed record NullableSystemTypeValue : SystemTypeValue
+	sealed record NullableSystemTypeValue : SingleValue
 	{
-		public NullableSystemTypeValue (in TypeProxy representedType, in MultiValue underlyingTypeValue) : base (representedType)
-			=> UnderlyingTypeValue = underlyingTypeValue;
+		public NullableSystemTypeValue (in TypeProxy nullableType, in TypeProxy underlyingType)
+		{
+			Debug.Assert (nullableType.Name == "Nullable`1" && nullableType.Namespace == "System");
+			UnderlyingTypeValue = underlyingType;
+			NullableType = nullableType;
+		}
+		public readonly TypeProxy NullableType;
 
-		public readonly MultiValue UnderlyingTypeValue;
+		public readonly TypeProxy UnderlyingTypeValue;
+	}
+
+	sealed record NullableValueWithDynamicallyAccessedMembers : ValueWithDynamicallyAccessedMembers
```

Comment:
The only reason we need this value type is that applying annotation to it needs to apply the annotation to both the Nullable<T> and the underlying type (otherwise treating this as the underlying value would work, since we transparently return the same value when asked for underlying type).
Could you please add a comment what is the purpose of this value node?

Reply:
Similar reason applies to the `NullableSystemTypeValue`.

Reply:
Yes, I added comments for those definitions.